sdccc/src/main/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ConditionalPreconditions.java

```diff
@@ -349,6 +352,189 @@ static boolean manipulation(final Injector injector) throws PreconditionExceptio
+                            .filter(abstractDescriptor ->
+                                    abstractDescriptor.getClass().equals(MdsDescriptor.class))
+                            .findAny();
+                    if (mdsDescriptorPresent.isPresent()) {
+                        switch (ImpliedValueUtil.getModificationType(reportPart)) {
+                            case CRT -> crtSeen.set(true);
+                            case UPT -> uptSeen.set(true);
+                            case DEL -> delSeen.set(true);
+                            default -> {}
+                        }
+                    }
+                }
+            } catch (IOException | RuntimeException e) {
+                throw new PreconditionException(
+                        "An error occurred while trying to process description modification report messages from storage",
+                        e);
+            }
+            // CHECKSTYLE.ON: IllegalCatch
+            return crtSeen.get() && uptSeen.get() && delSeen.get();
+        }
+
+        static boolean manipulation(final Injector injector) {
+            final var manipulations = injector.getInstance(Manipulations.class);
+            final var testClient = injector.getInstance(TestClient.class);
+
+            final MdibAccess mdibAccess;
+            final SdcRemoteDevice remoteDevice = testClient.getSdcRemoteDevice();
+            if (remoteDevice == null) {
+                LOG.error("remote device could not be accessed, likely due to a disconnect");
+                return false;
+            }
+            mdibAccess = remoteDevice.getMdibAccess();
+
+            final List<String> removableMdsDescriptors =
+                    manipulations.getRemovableDescriptorsOfClass(MdsDescriptor.class);
+            if (removableMdsDescriptors.isEmpty()) {
+                LOG.error("No removable MdsDescriptors could be found via the GetRemovableDescriptorsOfType "
+                        + "manipulation. Please check if the test case applying this precondition is applicable to "
+                        + "your device and if the GetRemovableDescriptorsOfType manipulation has been implemented "
+                        + "correctly.");
+            }
+
+            final List<String> absentRemovableMdsDescriptors = new ArrayList<>();
+            final List<String> presentRemovableMdsDescriptors = new ArrayList<>();
+            for (String removableMdsDescriptor : removableMdsDescriptors) {
+                final Optional<AbstractDescriptor> descOpt = mdibAccess.getDescriptor(removableMdsDescriptor);
+                if (descOpt.isEmpty()) {
+                    absentRemovableMdsDescriptors.add(removableMdsDescriptor);
+                } else {
+                    presentRemovableMdsDescriptors.add(removableMdsDescriptor);
+                }
+            }
+
+            // absent:        present:
+            // 1. insert
```

Comment:
Wouldn't it be easier to add all the removed ones back first and then apply the same procedure to all of them?

Reply:
Yes, that would be easier. But this way all Descriptors would either be present or be absent in the end.
I tried to design the Precondition in a way that does not change their presence.